openpype/lib/events.py

```diff
@@ -34,20 +234,32 @@ class EventCallback(object):
     or none arguments. When 1 argument is expected then the processed 'Event'
     object is passed in.
 
-    The registered callbacks don't keep function in memory so it is not
-    possible to store lambda function as callback.
+    The callbacks are validated against their reference counter, that is
+        achieved using 'weakref' module. That means that the callback must
+        be stored in memory somewhere. e.g. lambda functions are not
+        supported as valid callback.
+
+    You can use 'partial' functions. In that case is partial object stored
+        in the callback object and reference counter is checked for the
+        wrapped function.
```

Comment:
Hmm - yeah, not sure about this. Partials don't necessarily live as long as the function they reference. They are usually intended to be more shortlived than the function they reference (if they do intend to go out of scope that is).